serf/serf.go

```diff
@@ -25,7 +25,7 @@ import (
 // version to memberlist below.
 const (
 	ProtocolVersionMin uint8 = 2
-	ProtocolVersionMax       = 4
+	ProtocolVersionMax       = 5
```

Comment:
I'd have to dig through the code again to remember how this part works, but there's a similar mapping in Consul - https://github.com/hashicorp/consul/pull/2684.

That maps Consul protocol versions to Serf versions. We need to make sure we account for the 5 there properly (if at all).